### ISSUE TITLE
Support Group Normalization?

### ISSUE THREAD

**Issue by yaoliUoA** [facebookresearch/maskrcnn-benchmark#110]
## ❓ Questions and Help
Hi, 

Does this implementation supports group normailization? I think Detectron supports group normalization (https://github.com/facebookresearch/Detectron/blob/master/projects/GN/README.md). If not, any plans to add group normailzation in this package?

**Comment by fmassa** [comment]
Hi,

We don't have Group Normalization implemented yet, but it would be good to have.
I'm currently working on other parts of the library, but contributions are more than welcome! :-)

**Comment by zimenglan-sysu-512** [comment]
hi @fmassa 
do u have a plan to add GN? 
those [Detectron.pytorch](https://github.com/roytseng-tw/Detectron.pytorch), [Detectron-Cascade-RCNN](https://github.com/zhaoweicai/Detectron-Cascade-RCNN), and [PANet](https://github.com/ShuLiu1993/PANet), use python to implement it, while [Detectron](https://github.com/facebookresearch/Detectron) use caffe2. 
can u give me some ideas how to add GN to this benchmark?

**Comment by fmassa** [comment]
It should be fairly straightforward, like changing `FrozenBatchNorm2d` by `GroupNorm`, plus a few architectural changes. Basically, [change this file](https://github.com/facebookresearch/maskrcnn-benchmark/blob/master/maskrcnn_benchmark/modeling/backbone/resnet.py), adding the new blocks in the registration in https://github.com/facebookresearch/maskrcnn-benchmark/blob/83849b4ed116ee30a4a3d94519f7ee29d8755b2d/maskrcnn_benchmark/modeling/backbone/resnet.py#L294-L305

After that, it would also require loading a pre-trained imagenet model trained using GN.

Let me know if you need more information

**Comment by zimenglan-sysu-512** [comment]
thanks @fmassa 
i follow what u said to add GN, and right now i try to run some experiments.
thanks again

**Comment by fmassa** [comment]
Thanks, let me know how it goes, and if it works, could you send a PR adding this functionality?

**Comment by zimenglan-sysu-512** [comment]
hi @fmassa 
i add `GN` from [PANet](https://github.com/ShuLiu1993/PANet), and retrain [retinanet](https://github.com/facebookresearch/maskrcnn-benchmark/pull/102/) using R50, only boost 0.5%  mAP (from 36.4 to 36.9). and training my own dataset using GN drops a little. 
maybe some bugs in the code i add with GN.

**Comment by fmassa** [comment]
@zimenglan-sysu-512 Could you try training with Faster R-CNN R-50 with GN? This way we will factor out any possible changes due to the retinanet implementation.

**Comment by zimenglan-sysu-512** [comment]
hi @fmassa 
it took a week to run `Faster R-CNN R-50 with GN` from `scratch`, here is the results on coco
```
Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.369 
Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=100 ] = 0.584 
Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=100 ] = 0.400
Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.224
Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.397
Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.476
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=  1 ] = 0.310
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets= 10 ] = 0.491
Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.517 
Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=100 ] = 0.339
Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=100 ] = 0.549 
Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=100 ] = 0.649
```
i need some times to send a PR. (please wait)
thanks

**Comment by fmassa** [comment]
This is great, having an initial implementation will be awesome, and we can maybe dig to find the reason for the 2.6 mAP drop.

**Comment by fmassa** [comment]
Fixed via https://github.com/facebookresearch/maskrcnn-benchmark/pull/346
Thanks @zimenglan-sysu-512
